spec.md

```diff
@@ -2374,6 +2407,36 @@ SInt("h-2A")
 SInt("h-2a")
 ```
 
+## Enum Expressions
+
+An enumeration can be constructed by specifying the variant tag, an optional
+data value expression, and a return type specifier.
```

Comment:
Should probably clarify that the data value expression is only optional when the data type is a `UInt<0>`, and is implicitly defined as `UInt<0>(0)`.

Reply:
It can't be `UInt<0>(0)` though right since, 1) if you have multiple alternatives, they should have different values and 2) if undefined, the compiler/synthesis tool should be free to choose any value.

Reply:
i think the assumption is if they dont carry a value, they can never be converted to a hardware value / used in a register...? I'm not sure the use cases given the current set of features how it would be used to have multiple enums with the same 0.U value?

Reply:
My confusion was the use of `UInt<0>` as defined above as the implicit data type associated with tags that notionally do not have one. It might be worth calling that out even if it is slightly repetitive, eg. something like:
```
An enumeration can be constructed by specifying the variant tag, an optional
data value expression, and a return type specifier.
The data value is required unless the tag's associated data type is `UInt<0>`
which indicates there is no data associated with that tag.
```
Probably something better than that, but that's the idea I'm going for.

I agree with @youngar's proposed clarification.